stock_picking_invoicing_unified/wizard/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+# © 2016 Ainara Galdona <ainaragaldona@avanzosc.es> - Avanzosc S.L.
+# © 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields, api
+from openerp.tools import config
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+
+    _inherit = 'stock.invoice.onshipping'
+
+    journal_id = fields.Many2one(required=False)
+    sale_journal = fields.Many2one(
+        comodel_name='account.journal', string='Sale Journal',
+        domain="[('type', '=', 'sale')]")
```

Comment:
For curiosity: is there a good reason to use domain as a string, instead of sequence: domain=[('type', '=', 'sale')] ? I usually declare domains as a sequence in my Python code, not as a string... but I don't know if it is equivalent or not.